chapter04/chapter04.adoc

```diff
@@ -0,0 +1,317 @@
+
+The OpenAPI specification enables creation of a well-defined, clear and comprehensive API contract. It provides a standardized way to describe the API's structure, expected requests and responses, and authentication mechanisms, making it easier to develop, test, and maintain RESTful APIs.
+
+== Introduction to MicroProfile OpenAPI
+
+MicroProfile OpenAPI specification builds upon the widely recognized OpenAPI Specification (OAS) and leverages annotations from the Jakarta Restful Web Services specification. The primary focus of MicroProfile OpenAPI is on defining REST APIs that utilize JSON within the context of HTTP.
+
+The specification aims to provide a uniform way of describing APIs so that it is both human-readable and machine-readable.  It facilitates the creation of APIs that are consistent, well-documented, and easily consumable by both humans and machines. 
+
+== Capabilities of MicroProfile OpenAPI Specification
+
+MicroProfile OpenAPI provides a suite of Java APIs that allows developers to define and generate API specifications that adhere to OpenAPI v3 standards. As a result, it simplifies the process of designing, documenting, and publishing RESTful APIs for developers. 
+
+Developers can quickly generate documentation for their microservices using MicroProfile OpenAPI. The documentation includes information on what services are provided, how to invoke them, and what data types are used. It generates comprehensive metadata about services, ensuring interoperability across diverse platforms and tools. Also, documentation can generate client code to access the web services.
+
+This specification streamlines the creation of OpenAPI documentation for RESTful services using a unified approach. It generates comprehensive metadata about services, ensuring interoperability across diverse platforms and tools. The OpenAPI Specification fuels a rich ecosystem of tools that automate and support:
+
+* *API Documentation Generation*: Intuitive interactive documentation portals emerge directly from the specification.
+* *Client SDK Creation*: Client libraries in various languages can be automatically generated.
+* *API Testing*: Testing frameworks can leverage the specification to design robust tests.
+* *API Mocking*: Simplifies mocking APIs for testing and development purposes.
+
+=== Generating OpenAPI documents
+
+There are multiple ways in which you can generate OpenAPI documents. The most common way is to use annotations. This only requires augmenting your Jakarta Restful Web Services annotations with OpenAPI annotations. 
+
+You can also use JSON or YAML files. You can start with a predefined model and manually update the entire OpenAPI document tree of your web services. However, the annotation-based approach is recommended as it is more maintainable and easier to understand. Finally, you can filter out the resources you do not want to document using configuration.
```

Comment:
> You can also use JSON or YAML files. You can start with a predefined model and manually update the entire OpenAPI document tree of your web services.

This could be a little more clear. For example, something like "_Besides annotations, a predefined OpenAPI document may be provided in either YAML or JSON format. This so-called static model will be merged with the model generated by scanning for Jakarta REST endpoints and the combined result will be made available to clients_".

Reply:
Fixed with https://github.com/ttelang/microprofile-tutorial/commit/63f99ab89d0d99f29356ceea1c2f4dd5be43e1a4